### PR TITLE
Solidity 0.8.12

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: solidity.rb [hunk 1]
```diff
@@ -16,9 +16,9 @@
 class Solidity < Formula
   desc "The Solidity Contract-Oriented Programming Language"
   homepage "https://docs.soliditylang.org"
-  url "https://github.com/ethereum/solidity/releases/download/v0.8.9/solidity_0.8.9.tar.gz"
-  version "0.8.9"
-  sha256 "36643d0282998136b610f740808acb4dc6728144bb9ee70e5fd4124cda85ddab"
+  url "https://github.com/ethereum/solidity/releases/download/v0.8.9/solidity_0.8.11.tar.gz"
```
Comment (cameel) [inline solidity.rb:19]:
```suggestion
  url "https://github.com/ethereum/solidity/releases/download/v0.8.11/solidity_0.8.11.tar.gz"
```

**Comment by cameel** [comment]
Just wanted to push a fix for the timeout we saw with 0.8.10 but looks like I don't have permissions.

Anyway, I realized that we can just add something like `no_output_timeout: 30m` to the job in CircleCI config to increase the timeout. It this times out too, please add it.

**Comment by cameel** [comment]
Tests failed. Looks like it still needs the timeout fix.

**Comment by cameel** [comment]
Now there's a different error:
```
Error: Your Xcode (11.4.1) is outdated.
Please update to Xcode 12.4 (or delete it).
Xcode can be updated from the App Store.
```

I'll update it to the latest (13.2) like we did in Solidity repo.

**Comment by cameel** [comment]
And now compilation fails on 0.4.

This is what the log shows:
```
Applications/Xcode-13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
/usr/bin/ranlib libsolidity.a
/Applications/Xcode-13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
```
I think this is not the actual error that interrupted the compilation. It was probably higher up and not included in the output. So looks like this might require getting into the container and compiling manually.

The problem could be some new warning from clang so I'll try to switch back to an earlier Xcode version and see if it compiles on that.

**Comment by EmtMeshal** [comment]
BinKareem

**Pull request closed without merging**